### PR TITLE
Add missing docs to primitive_conversions library in standard library

### PR DESCRIPTION
**Pull request by GearedPaladin**
## Checklist

- [ ] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by GearedPaladin** [comment]
@crodas @bitzoic please review

**Review by sdankel — APPROVED** [review]
LGTM, thanks

File: sway-lib-std/src/primitive_conversions/u16.sw [hunk 1]
```diff
@@ -5,6 +5,27 @@ use ::option::Option::{self, *};
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 255_u16.try_as_u8();
+    ///     assert(val == Some(255_u8));
+    ///
+    ///     // Conversion fails as value is above the max a u8 can represent.
+    ///     let val2 = 256_u16.try_as_u8();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u16.sw:28]:
```suggestion
    /// }
    /// ```
```

File: sway-lib-std/src/primitive_conversions/u32.sw [hunk 2]
```diff
@@ -5,6 +5,27 @@ use ::option::Option::{self, *};
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 255_u32.try_as_u8();
+    ///     assert(val == Some(255_u8));
+    ///
+    ///     // Conversion fails as value is above the max a u8 can represent.
+    ///     let val2 = 256_u32.try_as_u8();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u32.sw:28]:
```suggestion
    /// }
    /// ```
```

File: sway-lib-std/src/primitive_conversions/u32.sw [hunk 3]
```diff
@@ -15,6 +36,27 @@ impl u32 {
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 65_535_u32.try_as_u16();
+    ///     assert(val == Some(65_535_u16));
+    ///
+    ///     // Conversion fails as value is above the max a u16 can represent.
+    ///     let val2 = 65_536_u32.try_as_u16();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u32.sw:59]:
```suggestion
    /// }
    /// ```
```

File: sway-lib-std/src/primitive_conversions/u64.sw [hunk 4]
```diff
@@ -5,6 +5,27 @@ use ::option::Option::{self, *};
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 255_u64.try_as_u8();
+    ///     assert(val == Some(255_u8));
+    ///
+    ///     // Conversion fails as value is above the max a u8 can represent.
+    ///     let val2 = 256_u64.try_as_u8();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u64.sw:28]:
```suggestion
    /// }
    /// ```
```

File: sway-lib-std/src/primitive_conversions/u64.sw [hunk 5]
```diff
@@ -15,6 +36,27 @@ impl u64 {
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 65_535_u64.try_as_u16();
+    ///     assert(val == Some(65_535_u16));
+    ///
+    ///     // Conversion fails as value is above the max a u16 can represent.
+    ///     let val2 = 65_536_u64.try_as_u16();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u64.sw:59]:
```suggestion
    /// }
    /// ```
```

File: sway-lib-std/src/primitive_conversions/u64.sw [hunk 6]
```diff
@@ -25,6 +67,27 @@ impl u64 {
...
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let val = 4_294_967_295_u64.try_as_u32();
+    ///     assert(val == Some(4_294_967_295_u32));
+    ///
+    ///     // Conversion fails as value is above the max a u32 can represent.
+    ///     let val2 = 4_294_967_296_u64.try_as_u32();
+    ///     assert(val == None);
+    /// }
```
Comment (bitzoic) [inline sway-lib-std/src/primitive_conversions/u64.sw:90]:
```suggestion
    /// }
    /// ```
```

**Review by bitzoic — COMMENTED** [review]
These are great! Just need to add the closing wrapper on the examples. Thank you for your contribution!

**Review by sdankel — APPROVED** [review]

**Review by SwayStar123 — APPROVED** [review]

**Pull request merged**